### PR TITLE
plugin: add EndpointResolver interface

### PR DESCRIPTION
**Pull request by thaJeztah**
related to https://github.com/moby/moby/pull/43298#discussion_r820972861

This defines the interface that the package expects in order to lookup pull endpoints, instead of requiring the whole registry.Service interface

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by corhere — APPROVED** [review]

**Comment by thaJeztah** [comment]
Oh! Hmm.. looks like I screwed something up. Will check when back at my computer 

```
INFO [runner] linters took 27.968139049s with stages: goanalysis_metalinter: 27.887976013s 
[2022-03-11T14:42:36.808Z] plugin/manager.go:98:18: SA5007: infinite recursive call (staticcheck)
[2022-03-11T14:42:36.808Z] 	repoInfo, err = s.ResolveRepository(name)
[2022-03-11T14:42:36.808Z] 	                ^
```

File: plugin/manager.go [hunk 1]
```diff
@@ -86,11 +91,11 @@ type controller struct {
 // pluginRegistryService ensures that all resolved repositories
 // are of the plugin class.
 type pluginRegistryService struct {
-	registry.Service
+	EndpointResolver
 }
 
 func (s pluginRegistryService) ResolveRepository(name reference.Named) (repoInfo *registry.RepositoryInfo, err error) {
-	repoInfo, err = s.Service.ResolveRepository(name)
+	repoInfo, err = s.ResolveRepository(name)
```
Comment (cpuguy83) [inline plugin/manager.go:98]: This is recursive? Also I don't think anything on `pluginRegistryService` implements `ResolveRepository` except itself.

Comment (thaJeztah) [reply plugin/manager.go]: Yes, I was coming to the same conclusion; just pushed a commit that removes the whole `pluginRegistryService` type

**Comment by thaJeztah** [comment]
@cpuguy83 @corhere all green now; ptal

**Review by corhere — APPROVED** [review]
LGTM!

**Review by tianon — APPROVED** [review]

File: plugin/manager.go [hunk 2]
```diff
@@ -83,25 +87,8 @@ type controller struct {
...
-// pluginRegistryService ensures that all resolved repositories
-// are of the plugin class.
-type pluginRegistryService struct {
-	registry.Service
-}
-
-func (s pluginRegistryService) ResolveRepository(name reference.Named) (repoInfo *registry.RepositoryInfo, err error) {
-	repoInfo, err = s.Service.ResolveRepository(name)
-	if repoInfo != nil {
-		repoInfo.Class = "plugin"
```
Comment (cpuguy83) [inline plugin/manager.go:95]: Is this being set somewhere else?

Comment (thaJeztah) [reply plugin/manager.go:95]: That's what I was wondering; this code didn't appear to be called though, so not by this code

Comment (thaJeztah) [reply plugin/manager.go:95]: I could double check (not near a computer), but at least couldn't find where it was called

Comment (thaJeztah) [reply plugin/manager.go:95]: Think that could relate to the failures on https://github.com/moby/moby/pull/42889 ?

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**